common/client/transaction_sender_test.go

```diff
@@ -0,0 +1,422 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"go.uber.org/zap"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/utils/tests"
+	"github.com/smartcontractkit/chainlink/v2/common/types"
+
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+)
+
+type sendTxMultiNode struct {
+	*MultiNode[types.ID, SendTxRPCClient[any]]
+}
+
+type sendTxMultiNodeOpts struct {
+	logger        logger.Logger
+	selectionMode string
+	leaseDuration time.Duration
+	nodes         []Node[types.ID, SendTxRPCClient[any]]
+	sendonlys     []SendOnlyNode[types.ID, SendTxRPCClient[any]]
+	chainID       types.ID
+	chainFamily   string
+}
+
+func newSendTxMultiNode(t *testing.T, opts sendTxMultiNodeOpts) sendTxMultiNode {
+	if opts.logger == nil {
+		opts.logger = logger.Test(t)
+	}
+
+	result := NewMultiNode[types.ID, SendTxRPCClient[any]](
+		opts.logger, opts.selectionMode, opts.leaseDuration, opts.nodes, opts.sendonlys, opts.chainID, opts.chainFamily)
+	return sendTxMultiNode{
+		result,
+	}
+}
+
+type sendTxRPC struct {
+	sendTxErr error
+}
+
+var _ SendTxRPCClient[any] = (*sendTxRPC)(nil)
+
+func newSendTxRPC(sendTxErr error) *sendTxRPC {
+	return &sendTxRPC{sendTxErr: sendTxErr}
+}
+
+func (rpc *sendTxRPC) SendTransaction(ctx context.Context, tx any) error {
+	return rpc.sendTxErr
+}
+
+func TestMultiNode_SendTransaction(t *testing.T) {
+	t.Parallel()
```

Comment:
Could you ensure that there is no flaky tests by running TxSender tests with `-count=1000`, after addressing all other comments?